### PR TITLE
ci: `CI.yaml`

### LINKED ISSUE #230: Introduce CI so that we confirm that PRs are correct before merging.
**Is your feature request related to a problem? Please describe.**
#229 illustrates a case where the build was broken requiring a fix.

**Describe the solution you'd like**
Build (but don't deploy) branches for PRs.

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
## What I did

Created `CI.yaml` in `workflows/` that runs the following upon a `push` and/or `pull request` to trunk:

- npm ci (clean install)
- npm run build
- npm run build:staging

closes #230

### CODE REVIEW

**Review by xavierchanth — APPROVED** [review]

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - name: clean install
+        run: npm ci
+      - name: build
```
Comment (xavierchanth) [inline .github/workflows/CI.yaml]: We don't need to build before calling build:staging, maybe instead use a matrix to test a build:staging and a build:production in parallel?

**Review by xavierchanth — CHANGES_REQUESTED** [review]

**Comment by xavierchanth** [comment]
The other check we need to do here is make sure the css isn't overpurged.
Maybe we can check the contents/length of hugo_stats.json after running the build.
If it is short(less than 50 lines)/empty/doesn't exist, then fail the CI.

**Review by xavierchanth — APPROVED** [review]
LGTM

**Comment by JeremyTubongbanua** [comment]
Ready to merge @XavierChanth

**Pull request merged**